Compilation error when using newer version of servant

I updated tho the latest LTS: 12.24; I then ran into this issue:

```
/src/App.hs:50:8: error:                                                                                                                        
    * Couldn't match expected type `Handler Item'                                                                                                                                  
                  with actual type `ExceptT ServantErr m0 a0'                                                                                                                      
    * In the expression: throwE err404                                                                                                                                             
      In a case alternative: _ -> throwE err404                                                                                                                                    
      In the expression:                                                                                                                                                           
        \case                                                                                                                                                                      
          0 -> return exampleItem                                                                                                                                                  
          _ -> throwE err404                                                                                                                                                       
   |                                                                                                                                                                               
50 |   _ -> throwE err404                                                                                                                                                          
   |        ^^^^^^^^^^^^^  
```

--- a/src/App.hs
+++ b/src/App.hs
@@ -5,7 +5,6 @@
 
 module App where
 
-import           Control.Monad.Trans.Except
 import           Data.Aeson
 import           GHC.Generics
 import           Network.Wai
@@ -47,7 +46,7 @@ getItems = return [exampleItem]
 getItemById :: Integer -> Handler Item
 getItemById = \ case
   0 -> return exampleItem
-  _ -> throwE err404
+  _ -> throwError err404
 
 exampleItem :: Item
 exampleItem = Item 0 "example item"

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,3 +1,3 @@
-resolver: lts-11.0
+resolver: lts-13.8
 packages:
   - '.'

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,3 +1,3 @@
-resolver: lts-8.12
+resolver: lts-11.0
 packages:
   - '.'

--- a/test/AppSpec.hs
+++ b/test/AppSpec.hs
@@ -27,15 +27,20 @@ spec = do
         try env (getItem 0) `shouldReturn` Item 0 "example item"
 
       it "throws a 404 for missing items" $ \ env -> do
-        try env (getItem 42) `shouldThrow` (\ e -> responseStatus e == notFound404)
+        try env (getItem 42) `shouldThrow` errorsWithStatus notFound404
+
+errorsWithStatus :: Status -> ServantError -> Bool
+errorsWithStatus status servantError = case servantError of
+  FailureResponse response -> responseStatusCode response == status
+  _ -> False
 
 withClient :: IO Application -> SpecWith ClientEnv -> SpecWith ()
 withClient x innerSpec =
   beforeAll (newManager defaultManagerSettings) $ do
-    flip aroundWith innerSpec $ \ action -> \ manager -> do
+    flip aroundWith innerSpec $ \ action -> \ httpManager -> do
       testWithApplication x $ \ port -> do
-        let baseUrl = BaseUrl Http "localhost" port ""
-        action (ClientEnv manager baseUrl)
+        let testBaseUrl = BaseUrl Http "localhost" port ""
+        action (ClientEnv httpManager testBaseUrl Nothing)
 
 type Host = (Manager, BaseUrl)
 

